include/modem/location.h

```diff
@@ -62,7 +71,12 @@ enum location_event_id {
 	 * GNSS is requesting P-GPS data. Application should obtain the data and send it to
 	 * location_pgps_data_process().
 	 */
-	LOCATION_EVT_GNSS_PREDICTION_REQUEST
+	LOCATION_EVT_GNSS_PREDICTION_REQUEST,
+	/**
+	 * Neighbor cell information is available. Application should send the cell information to
+	 * cloud services and then pass the result to location_cellular_ready().
+	 */
+	LOCATION_EVT_NEIGHBOR_CELL_REQUEST
```

Comment:
I'm not sure whether to use cellular or neighbor cell in naming of event, Kconfig, variable in location_event_data and function name. Now this event name is only one with neighbor cell and the rest are named with cellular.

Reply:
I don't have a good answer tbh. The neighbor cell request is based on neighbor cell measurements, but a request can also only include only the current cell. So sounds like cellular encapsulated both the best? Does the event only propagate if neighbor cells are found?

Reply:
The serving cell may also be the only cell we get and that's still enough to resolve the location of the device. I think neighbor cells are more like a bonus, which we may get and which may improve the accuracy. So, I would rather use the term _cellular_ or _cellular location_. The current naming probably comes from the used AT command, but I don't think we have to care about that.

Reply:
Good point on neighbors not present always so I'll change this to cellular, although I've always though that current cell is just a neighbor from location perspective :)

Reply:
Note that we also use the term `neighbor cell` throughout the application. For instance the cloud codec API
```
int cloud_codec_encode_neighbor_cells(struct cloud_codec_data *output,
				      struct cloud_data_neighbor_cells *neighbor_cells);
```

The naming should ideally be aligned across the entire application. Maybe not in this PR, as the changes seem to pile up. Ill leave that up to you

Reply:
Renamed `LOCATION_EVT_NEIGHBOR_CELL_REQUEST` to `LOCATION_EVT_CELLULAR_REQUEST`.

I won't be doing any other `neighbor_cell(s)` to `cellular` renaming because that'll hide the main point of this PR. And I'm not sure if that's something that would even need to be done.